### Tugas Analisis 1:
Ketika mencoba menggunakan list Python biasa, program akan error dan menampilkan kesalahan TypeError: can't multiply sequence by non int of type 'float'. Tapi jika menggunakan Array program akan mengalikan tiap angka di dalam array, tidak di jadikan satu


In [7]:
#1. Operasi dasar Numpy Array
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])
print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga Terendah :', harga.min())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga Terendah : 3000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


### Jawaban 1:
Ketika mencoba menggunakan list Python biasa, program akan error dan menampilkan kesalahan TypeError: can't multiply sequence by non int of type 'float'. Tapi jika menggunakan Array program akan mengalikan tiap angka di dalam array, tidak di jadikan satu

---






### Tugas Analisis 2:
Amati tabel yang dihasilkan. Kolom mana yang terlihat memiliki data kosong
(None)? Menurutmu apa risikonya jika data ini langsung dianalisis tanpa dibersihkan?

In [8]:
#2. Membuat Series & DataFrame
import pandas as pd

data_kantin = {
'menu':      ['Nasi Goreng' , 'Es Teh'  , 'Mie Ayam', 'Es Teh'  , None  ],
'harga':     [12000         , 4000      , 10000     , 4000      , 8000  ],
'terjual':   [23            , 40        , None      , 35        , 18    ]
}

df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4          NaN   8000     18.0


### Jawaban 2:
Kolom yang tidak memiliki data adalah kolom "menu" di index ke 4 dan "terjual" di index ke 2 . Resikonya adalah hasil perhitungan tidak akan akurat, Perhitungan statistik seperti rata-rata unit terjual dapat menjadi tidak konsisten

---

### Tugas Analisis 3:
Dari hasil df.info(), kolom mana yang jumlah non-null-nya lebih sedikit dari jumlah baris total? Apa artinya?

In [9]:
#3. Data Loading & Inspection

df = pd.read_csv("data_kantin.csv")

print(df.head())        # 5 Baris Pertama
print()

print(df.info())        # tipe data dan jumlah non-null
print()

print(df.describe())    # statistik ringkas kolom numerik
print()

print("Jumlah baris dan kolom adalah : ", df.shape)         # jumlah (baris dan kolom)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0

<class 'pandas.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    str    
 1   menu      107 non-null    str    
 2   kategori  110 non-null    str    
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), str(3)
memory usage: 4.4 KB
None

              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000   33.000000
m

### Jawaban 3:
kolom yang memiliki jumlah non-null-nya lebih sedikit dari jumlah baris total adalah kolom "terjual". Artinya kolom terjual memiliki 6 baris/index data yang kosong

---

### Tugas Analisis 4:
Mengapa pada contoh di bawah kolom terjual diisi (fillna) sedangkan baris
dengan menu kosong justru dihapus (dropna)? Diskusikan alasan logisnya.

In [10]:
#4. Menangani Missing Value
print(df.isnull().sum()) # jumlah data kosong tiap kolom

df['terjual'] = df['terjual'].fillna(0) # isi kekosongan dengan 0
df = df.dropna(subset=['menu']) # hapus baris jika kolom menu kosong

print(df.head())

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64
      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0


### Jawaban 4:
Data "Terjual" di isi 0 karena angka Nol dan non-null bisa diartikan sebagai tidak ada penjualan. Sedangkan baris dengan menu kosong dihapus karena transaksi tanpa identitas produk merusak akurasi data

---

### Tugas Analisis 5:
Setelah drop_duplicates(), berapa jumlah baris dataset sekarang dibanding
sebelumnya? Mengapa penting memastikan tipe data (dtypes) sudah benar sebelum data
dianalisis lebih lanjut?

In [11]:
#5. Menangani Duplikat dan Tipe Data
print(df.duplicated().sum()) # jumlah baris duplikat
print(df.info())
df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int) # memastikan tipe data harga adalah integer
print(df.dtypes)

print(df.info())

5
<class 'pandas.DataFrame'>
Index: 107 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   107 non-null    str    
 1   menu      107 non-null    str    
 2   kategori  107 non-null    str    
 3   harga     107 non-null    int64  
 4   terjual   107 non-null    float64
dtypes: float64(1), int64(1), str(3)
memory usage: 5.0 KB
None
tanggal         str
menu            str
kategori        str
harga         int64
terjual     float64
dtype: object
<class 'pandas.DataFrame'>
Index: 102 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   102 non-null    str    
 1   menu      102 non-null    str    
 2   kategori  102 non-null    str    
 3   harga     102 non-null    int64  
 4   terjual   102 non-null    float64
dtypes: float64(1), int64(1), str(3)
memory usage: 4.8 KB
None


### Jawaban 5:
Setelah drop_duplicates(), jumlah baris dataset sekarang jadi 102 dari yang sebelumnya 107, "Kok bisa 102?" itu dikarenakan 3 baris nilai kosong di kolom "menu" di hapus dengan "dropna".
Pentingnya memastikan tipe data adalah agar Akurasi perhitungan Agregasi.

---

### Tugas Analisis 6
Dari hasil groupby di bawah, menu apa yang menghasilkan total_pendapatan
tertinggi? Bagaimana informasi ini bisa membantu pengambilan keputusan di kantin sekolah?

In [12]:
#Data Manipulation (Filtering, Sorting, Groupby)
laris = df[df['terjual'] > 20] # filtering
urut = df.sort_values(by='terjual', ascending=False) # sorting

df['total_pendapatan'] = df['harga'] * df['terjual'] # kolom turunan

ringkasan = df.groupby('menu')['total_pendapatan'].sum() # agregasi
print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


### Jawaban 6
Dari hasil groupby, menu yang menghasilkan total_pendapatan
tertinggi adalah menu "Nasi Goreng" dengan pendapatan 3.924.000. informasi ini bisa membantu pengambilan keputusan pengelola kantin Memanejemen Stok bahan baku. 

---